testing/sdk_auth.py

```diff
@@ -191,16 +191,14 @@ def _install_marathon_app(app_definition):
             return success
 
         if sdk_marathon.app_exists(self.app_definition["id"]):
-            log.info("Found installed KDC app")
-            return _get_kdc_task(self.app_definition["id"])
+            log.info("Found installed KDC app, destroying it first")
+            sdk_marathon.destroy(self.app_definition["id"])
```

Comment:
Why was this change required? We need persistence for the KDC app.

It's a bit roundabout, but basically the `kdc.py` tooling that we use in soak (and for testing) imports `sdk_auth`. And we want the **same** KDC app to be used for multiple runs of the tool.

Reply:
This was mainly a problem when cleanup() isn't correctly called, e.g. if a test run is aborted. The following run would fail due to the stale kdc.

Would it make sense to start fresh each time for CI tests, but to reuse prior instances for soak tests?

Reply:
Let's make the default behaviour the cleanup behaviour and I can add the current functionality when using `kdc.py`. IIRC we pin to a SHA for soak, so we could do it as a follow-up (feel free to create a ticket if you think it's not part of this PR).

Reply:
To be clear, should the change being made here be retained, or should it be reverted to the prior/current behavior? (I'm fine with either)